lib.nix

```diff
@@ -82,6 +82,38 @@ let
     else if lib.strings.hasInfix "manylinux2014" f then { pkg = [ ml.manylinux2014 ]; str = "2014"; }
     else { pkg = [ ]; str = null; };
 
+  # Fetch the wheels from the PyPI index.
+  # We need to first get the proper URL to the wheel.
+  # Args:
+  #   pname: package name
+  #   file: filename including extension
+  #   hash: SRI hash
+  #   kind: Language implementation and version tag
+  fetchWheelFromPypi = lib.makeOverridable (
+    { pname, file, hash, kind }:
+    let
+      version = builtins.elemAt (builtins.split "-" file) 2;
+    in
+      pkgs.stdenv.mkDerivation {
+        name = file;
+        buildInputs = with pkgs; [ curl jq ];
+
+        phases = ["buildPhase"];
+        preferLocalBuild = true;
+        impureEnvVars = [
+          "http_proxy" "https_proxy" "ftp_proxy" "all_proxy" "no_proxy"
+        ];
+
+        buildPhase = ''
+          curl -k $(curl -k https://pypi.org/pypi/${pname}/json | jq -r '.releases."${version}"[] | select(.filename == "${file}") | .url') --output $out
```

Comment:
Could we try the calculated url and only then fall back to round-tripping to the API?

Reply:
@adisbladis I can do this

Reply:
Done